File: spring_java_orm_sample/src/main/java/com/example/demo/api/controller/PurchaseOperationApiController.java
package com.example.demo.api.controller;

import com.example.demo.api.PurchaseOperationApi;
import com.example.demo.api.request.PurchaseInvoiceCreateRequest;
import com.example.demo.api.response.PurchaseInvoiceResponse;
import com.example.demo.service.PurchaseOperationService;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@Validated
public class PurchaseOperationApiController implements PurchaseOperationApi {
    private final PurchaseOperationService service;

    @Override
    public PurchaseInvoiceResponse createPurchaseInvoice(PurchaseInvoiceCreateRequest request) {
        return service.create(request);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/api/controller/PurchaseOperationApiController.kt
package com.example.demo.api.controller

import com.example.demo.api.PurchaseOperationApi
import com.example.demo.api.request.PurchaseInvoiceCreateRequest
import com.example.demo.api.response.PurchaseInvoiceResponse
import com.example.demo.service.PurchaseOperationService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.RestController

@RestController
@Validated
class PurchaseOperationApiController(private val service: PurchaseOperationService) : PurchaseOperationApi {
    override fun createPurchaseInvoice(request: PurchaseInvoiceCreateRequest): PurchaseInvoiceResponse {
        return service.create(request)!!
    }
}
